### ISSUE TITLE
Bazel RBE: end2end:end2end_test@poller=poll-cv, tests passed but timeout reported

### ISSUE THREAD

**Issue by apolcyn** [grpc/grpc#17493]
https://source.cloud.google.com/results/invocations/f8a7db7b-925e-4596-b5ea-75f2b4f2b20b/targets/%2F%2Ftest%2Fcpp%2Fend2end:end2end_test@poller%3Dpoll-cv/tests

noticed on PR: https://github.com/grpc/grpc/pull/17470

at the bottom of the log:

```
D1212 21:37:49.603974492      20 end2end_test.cc:227]        TestScenario{use_interceptors=true, use_proxy=false, inproc=false, server_type=sync, credentials='INSECURE_CREDENTIALS'}
I1212 21:37:49.604221521      20 server_builder.cc:280]      Synchronous server. Num CQs: 4, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10
D1212 21:37:49.605185711      20 dns_resolver.cc:275]        Start resolving.
I1212 21:37:49.606475082      20 subchannel.cc:841]          New connected subchannel at 0x1357a60 for subchannel 0x7f04ac003020
[       OK ] ResourceQuotaEnd2end/ResourceQuotaEnd2endTest.SimpleRequest/3 (1006 ms)
[ RUN      ] ResourceQuotaEnd2end/ResourceQuotaEnd2endTest.SimpleRequest/4
D1212 21:37:50.609283643      20 ev_posix.cc:170]            Using polling engine: poll-cv
D1212 21:37:50.609437755      20 dns_resolver.cc:334]        Using native dns resolver
D1212 21:37:50.609539637      20 end2end_test.cc:227]        TestScenario{use_interceptors=false, use_proxy=false, inproc=true, server_type=sync, credentials='INSECURE_CREDENTIALS'}
I1212 21:37:50.609715228      20 server_builder.cc:280]      Synchronous server. Num CQs: 4, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10
[       OK ] ResourceQuotaEnd2end/ResourceQuotaEnd2endTest.SimpleRequest/4 (1003 ms)
[ RUN      ] ResourceQuotaEnd2end/ResourceQuotaEnd2endTest.SimpleRequest/5
D1212 21:37:51.611586954      20 ev_posix.cc:170]            Using polling engine: poll-cv
D1212 21:37:51.611697299      20 dns_resolver.cc:334]        Using native dns resolver
D1212 21:37:51.611789341      20 end2end_test.cc:227]        TestScenario{use_interceptors=true, use_proxy=false, inproc=true, server_type=sync, credentials='INSECURE_CREDENTIALS'}
I1212 21:37:51.611895582      20 server_builder.cc:280]      Synchronous server. Num CQs: 4, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10
[       OK ] ResourceQuotaEnd2end/ResourceQuotaEnd2endTest.SimpleRequest/5 (1003 ms)
[----------] 6 tests from ResourceQuotaEnd2end/ResourceQuotaEnd2endTest (6039 ms total)
[----------] Global test environment tear-down
[==========] 448 tests from 5 test cases ran. (447756 ms total)
[  PASSED  ] 448 tests.
```

**Comment by jtattermusch** [comment]
@billfeng327  links to a few more recent occurrences would be useful.
e.g. https://source.cloud.google.com/results/invocations/064156b6-799c-4e83-bf67-b82e572d9aaf is the most recent occurrence. 
- are there any obvious patterns?
- does it reproduce with a manual RBE run?

It is also worth noting that this is failing about 15% of RBE dbg builds, and it's our most common flake on RBE.

Btw, in the historical data, I also see (perhaps related) flakes for the same test, but this time with epoll1: `//test/cpp/end2end:end2end_test@poller=epoll1` and they are marked as "failure" instead of "timeout": https://source.cloud.google.com/results/invocations/4674c508-6537-4efd-95cd-7d60c8348341

**Comment by jtattermusch** [comment]
I was looking at the test data and I have a theory:
- poll-cv poller is known to be much slower than other pollers, so tests with poll-cv generally run longer
- even for run where all tests passed, the target `//test/cpp/end2end:end2end_test@poller=poll-cv` takes ~440s to run
- for failed targets the timeout seems to be after ~450 secs

Is the test timeout for this particular target just set too low?

**Comment by billfeng327** [comment]
Recent occurrences:

https://source.cloud.google.com/results/invocations/23a7880e-27a3-4883-b51c-be24f1beba57
https://source.cloud.google.com/results/invocations/dcfd6a3a-a535-4b82-b530-14f86b6ac8fc
https://source.cloud.google.com/results/invocations/816350e7-35e2-4a1d-887c-eab6265c54e9

all of which have poller poll-cv

**Comment by apolcyn** [comment]
From that, it looks like the timeout is after the individual test cases finish, likely due to extra time taken before `grpc_shutdown` at the end of main

**Comment by jtattermusch** [comment]
I found where the 450sec timeout is coming from:
- the test size is set to "medium" (which is the default)
- the canonical bazel command shows argument (`--test_timeout=300,450,1200,3600` in https://source.cloud.google.com/results/invocations/064156b6-799c-4e83-bf67-b82e572d9aaf/details)

The timeouts are being set in the `tools/internal_ci` shell scripts that execute RBE runs.

**Comment by apolcyn** [comment]
just saw this again in https://source.cloud.google.com/results/invocations/eb2702c0-907a-4edd-8e10-5d89231ab4f2/targets/%2F%2Ftest%2Fcpp%2Fend2end:end2end_test@poller%3Dpoll-cv/log

PR: https://github.com/grpc/grpc/pull/17818

**Comment by jtattermusch** [comment]
Should be closed by #17820 .
